### PR TITLE
Marcelo week 5 assessment

### PR DESCRIPTION
**Pull request by celoazevedo**

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -17,45 +17,112 @@
...
+
+const codedMessage = (str) => {
+    return str.split('').map(item =>
+        item === 'a' || item === 'A' ? item = '4' : 
+        item === 'e' || item === 'E' ? item = '3' : 
+        item === 'i' || item === 'I' ? item = '1' : 
+        item === 'o' || item === 'O' ? item = '0' : 
+        item
+    ).join('')
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:51]: Clean, readable, useful!!! Three cheers!

File: code-challenges.test.js [hunk 2]
```diff
@@ -17,45 +17,112 @@
...
+    // create a function that takes in an array as an argument
+    // iterate through the array accessing each element -- maybe able to use .filter()
+    // humm.. i wonder if JS has a .include? method (thinking in Ruby right now...) 
+        //! and JS does have an .includes() method. Also added an OR condition inside the filter to check for capital letter A
+    // ... this way we can use the method to check if the word includes the letter a or A
+    // and return an array with only the words that have a letter a or A
 
+const countainLetterA = (arr) => {
+    return arr.filter((item) => item.includes('a') || item.includes('A'))
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:80]: perfection!

File: code-challenges.test.js [hunk 3]
```diff
@@ -17,45 +17,112 @@
...
+
+    ///// initial approach....
+    // use a counter to check the number of instances of each element
+    // use the value from the counter (ex: counter1 = 3, and counter2 = 2) to check if we have a full house! 
+
+    ///// ... i found this snipet online in which we are declaring a variable counts and assigning an empty object. we apply the forEach method to iterate through the array and for every item in the array we store the item as a key in the counts object and we count the occurances of the item and store as the value of the key. 
+    // now I can check for the key values in the counts object (Object.values(counts)) --> this returns an array. 
+    // we can iterate through the values array (numOfsameItemsArr) for the presence of an element with the value of 2 and the value of 3!!!
+    // - instead of iterating, I used the bulit in method .includes to check for the presence of the elements 2 and 3 in the array.
+    // return true is the array includes 2 and includes 3. else return false
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:114]: Great explanation! I've seen a few people use this snippet so it's nice to see an explanation of it!

File: interview-questions.md [hunk 4]
```diff
@@ -7,53 +7,53 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 3. What is object-relational mapping?
 
-  Your answer:
+  Your answer: ORM allows us to write postgresql queries in the Rails console. It translates SQL into Rails and Rails into SQL. 
 
-  Researched answer:
+  Researched answer: We use the ORM -> Active Record framework in our Ruby on Rails application. It allows the developer to comunicate with the database using Ruby code in the Rails console (instead of manually writing the sql query statement). AAAND it lets you interact with the data as if it is a normal Ruby object.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:28]: Excellent!

File: interview-questions.md [hunk 5]
```diff
@@ -7,53 +7,53 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 5. What are primary keys? Why are they important?
 
-  Your answer:
+  Your answer: The primary key is a unique identifier for an instance on a database. It is used to make relations between two different data tables. Ex: We have a table of students and each student can have many books (we will also have a table for books). Every book in the books table will be connected to a student in the students table by a unique identifier / primary key (it is always in the belong_to side!). 
 
-  Researched answer:
+  Researched answer: a primary key is a specific choice of a column (attribute) that specifies a row in ralation with table. --> "which attributes identify a record"
```
Comment (WalkerTexas-coder) [inline interview-questions.md:44]: Both answers are on point!

File: rails-questions.md [hunk 6]
```diff
@@ -1,30 +1,37 @@
...
+    - $ rails new < app-name > -d postgresql -T
+        - here the -d flag is uninstalling the default database that comes with starting the new application with Rails and installing PostgreSQL. The -T flag is removing the default testing framework because we test with RSpec! And we install RSpec after we create our database but before we start creating our models. 
 
 4. What is the command to generate a Rails model for a meals table with columns named breakfast, lunch, and dinner?
-
+    - $ rails generate model Meal breakfast:string lunch:string dinner:string
 
 5. What is a migration? Why would you use one?
-
+    - Is the tool that we use to change the shape of our database. We can update the structure (adding, removing and changing column names) of our database by using migration. After we generate the model we need to migrate in order to create/update our schema and create a file for that model that was migrated. Every migration file will have a timestamp on its label and this will help us keep track of the changes (migrations) in our database.
```
Comment (WalkerTexas-coder) [inline rails-questions.md:18]: Great answer!

File: rails-questions.md [hunk 7]
```diff
@@ -1,30 +1,37 @@
...
+    - but before we migrate, we need something to migrate. We need to create our database and our models. By migrating we will start to create our tables!
 
 7. What is the naming convention for generating a Rails model? What is the naming convention for the table that is generated with this command?
-
+    - Rails model name convention -> PascalCasing and singluar
+    - Table that is generated upon the creation of the model -> snake_cased and plural
 
 8. What is the schema file in a Rails application? Can you modify the schema directly? Why or why not?
-
+    - The schema file have the structure of our database (our tables). We shoud not change the schema file directly. Rails is very opinionated and it likes to do everything for us. If we change the schema directly it will make Rails mad....
```
Comment (WalkerTexas-coder) [inline rails-questions.md:29]: haha and rails will break!

**Review by WalkerTexas-coder — APPROVED** [review]
MARCELOOOO!!! Excellent work! I really appreciate your use of ternary operators. They are a favorite tool of mine. Your pseudo code is excellent and and your ability to manipulate data and utilize the right tools for the job is perfect! I think your interview answers are fantastic and will be helpful as a resource as you go into interviews. Great job with the rails questions too!

**Comment by celoazevedo** [comment]
Hey Austin! Thank you for taking the time to review my assessment. I really appreciate all of your comments and feedback.

**Pull request merged**